docs/reference/ml/ml-shared.asciidoc

```diff
@@ -1,3 +1,27 @@
+tag::adaptive-allocation[]
+Adaptive allocations configuration object.
+If enabled, the number of allocations of the model is set based on the current load the process gets.
+When the load exceeds 90% of the current capacity, a new model allocation is automatically created (respecting the value of `max_number_of_allocations` if it's set).
+When the load is less than 85% of the current capacity, a model allocation is automatically removed (respecting the value of `min_number_of_allocations` if it's set).
```

Comment:
Regarding the 90% / 85%: I think I would keep it a bit more vague here. E.g. "When the load is high, ..." / "When the load is low ...". These numbers may change in the future and I'd rather have them hardcoded in the docs.